### DISCUSSION TITLE
LogBack support for syslog over TLS

### DISCUSSION THREAD

**Discussion by sharet-adl** [qos-ch/logback#588] [General]
Hi,
I am interested in how we could have logback configured to send syslog events over TLS, if possible. What I checked so far is:
- logback [Syslog appender](https://logback.qos.ch/manual/appenders.html#SyslogAppender) - which does not support TLS
- logback [SSLSocketAppender](https://logback.qos.ch/manual/appenders.html#SocketAppender) - which does not support Syslog 

I know that log4j2 supports syslog over tls, by using [SyslogAppender](https://logging.apache.org/log4j/2.x/manual/appenders.html#SyslogAppender).

What would be the best the options here, for LogBack ?

Thank you !

**Comment by sharet-adl** [comment]
For anyone interested in this, it seems the current best solution is to use an Appender from Logstash which is fully compatible with Logback and has support for TCP/TLS, namely [LogstashTcpSocketAppender](https://javadoc.io/doc/net.logstash.logback/logstash-logback-encoder/3.4/net/logstash/logback/appender/LogstashTcpSocketAppender.html). It would be nice if in the Logback documentation these sort of external and compatible Appenders that cover missing functionality in Logback could be mentioned.

**Reply by ceki** [reply]
FYI, logstash-logback-encoder is the first entry under "Third-party tools, extensions or forks" section on the [download page](https://logback.qos.ch/download.html)
